opm/material/fluidsystems/BlackOilFluidSystem.hpp

```diff
@@ -96,6 +96,15 @@ namespace FluidSystems {
  *
  * \tparam Scalar The type used for scalar floating point values
  */
+
+/*! Enum defining state of rate limmiting of dissolution to none, only with free phase or all
+ */
+enum RateLimmitCells {
+    None = 0,
+    Free = 1,
+    All  = 2
+};
```

Comment:
this enum definitely needs to bite the dust. opm-material is not  concerned with cells or even degrees of freedom. (Always remember that opm-material should be useful for solving ODEs for a chemical application.)